util/deephash/types.go

```diff
@@ -7,11 +7,20 @@ import (
 	"net/netip"
 	"reflect"
 	"time"
+
+	"tailscale.com/util/hashx"
 )
 
+// SelfHasher is the interface implemented by types that can compute their own hash
+// by writing values through the given parameter.
+type SelfHasher interface {
+	Hash(*hashx.Block512)
```

Comment:
(We also don't need to finalize the API now. I'm fine with it as is. We can clean this up later.)

Reply:
Makes sense - it would be good if we can newtype the `Block512` and just expose the basic hash-extend methods, but for now I've updated the doc string to make it super clear that this is unstable and not to call Reset/Sum.